### PR TITLE
Change the way of integrating the background rate in MapMaker

### PR DESCRIPTION
**Pull request by JouvinLea**
- use now the new method integration of background 2D and background3D based on the trapezoidal integration.
- we saw that f we do a trapezoidal integration compare to the rectangular one, the value of the test implemented for the integration doesn't change that much

**Questions that remain:**
- How many integration steps do we put by default? For the moment this is one but it has to be changed. Do we fix it or do we let it as a parameter of the MapMaker method?
- The test is not easy to understand and thus it is not easy to see that the integration is really working. For the moment there is only a test for background3D but not background 2D in the MapMaker. This is maybe just something we have to remember to improve but I must admit that I will not have time to improve this before leaving in holiday.

### CODE REVIEW

**Comment by cdeil** [comment]
@registerrier (and maybe @AtreyeeS ) - Thoughts?

Where do we want the simple evaluate at log center energy times energy bin width?
And where do we want the more accurate (but also slower / more complex code) log-log-trapz integral?

I think maybe just merge this PR, and `n_integration_bins` is a good value.
If we start to expose such values from high-level classes like the MapMaker, it gets too complex?
Basically we do something, and then what the user can choose is the energy binning to run the analysis, going finer for more accurate results?

**Comment by registerrier** [comment]
Some unrelated fails in the tests. Merging now.
